Can't compile with no tpls build

I am getting the following compile error with a fresh Tribo/mfem build.

````
src/tribol/mesh/MfemData.cpp:306:46: error: no member named 'Device' in 'tribol::MemorySpace'
  306 |       use_device_{ mem_space == MemorySpace::Device },
````

It's seems that `MemorySpace::Device` is used here

https://github.com/LLNL/Tribol/blob/5f9167f7ec3f8be3b19ca566465e6a064515de54/src/tribol/mesh/MfemData.cpp#L303

even though `TRIBOL_USE_UMPIRE` is not defined.

If it helps the most recent commit that has no issues is `aa7679e537622046776d74b57c7eefeff9d36611`.

--- a/src/tribol/mesh/MfemData.cpp
+++ b/src/tribol/mesh/MfemData.cpp
@@ -300,7 +300,7 @@ MfemMeshData::MfemMeshData( IndexT mesh_id_1, IndexT mesh_id_2, const mfem::ParM
       lor_factor_{ 0 },
       exec_mode_{ exec_mode },
       mem_space_{ mem_space },
-      use_device_{ mem_space == MemorySpace::Device },
+      use_device_{ isOnDevice( exec_mode ) },
       allocator_id_{ getResourceAllocatorID( mem_space ) }
 {
   // make sure a grid function exists on the submesh

